lib/model/model.go

```diff
@@ -1484,31 +1491,43 @@ func (m *Model) numHashers(folder string) int {
 // clusterConfig returns a ClusterConfigMessage that is correct for the given peer device
 func (m *Model) clusterConfig(device protocol.DeviceID) protocol.ClusterConfigMessage {
 	cm := protocol.ClusterConfigMessage{
+		DeviceName:    m.deviceName,
 		ClientName:    m.clientName,
 		ClientVersion: m.clientVersion,
-		Options: []protocol.Option{
-			{
-				Key:   "name",
-				Value: m.deviceName,
-			},
-		},
 	}
 
 	m.fmut.RLock()
 	for _, folder := range m.deviceFolders[device] {
+		folderCfg := m.cfg.Folders()[folder]
 		cr := protocol.Folder{
-			ID: folder,
+			ID:              folder,
+			ReadOnly:        folderCfg.ReadOnly,
+			RescanIntervalS: folderCfg.RescanIntervalS,
+			IgnorePerms:     folderCfg.IgnorePerms,
+			AutoNormalize:   folderCfg.AutoNormalize,
```

Comment:
I don't think we gain much by exposing AutoNormalize; it's more of an internal fixup thing in my mind than something that should be exposed and documented at the protocol level...